### PR TITLE
[doc] Add sparse grid example.

### PR DESCRIPTION
**Pull request by chunleili**
Issue: #

### <samp>🤖 Generated by Copilot at c39432b</samp>

This change updates the documentation with a new section on sparse data structures. It demonstrates how to use the `bitmasked` node to create a sparse grid that saves memory and adapts to the data.

### <samp>🤖 Generated by Copilot at c39432b</samp>

* Add a new section to the documentation on sparse data structures ([link](https://github.com/taichi-dev/taichi/pull/7858/files?diff=unified&w=0#diff-e951f0e114991e0acc29a038e8b18a0ebda9a9b58f035ebd9b75f6216a9f01a8R366-R405))

### CODE REVIEW

File: docs/lang/articles/basic/sparse.md [hunk 1]
```diff
@@ -363,6 +363,48 @@ print(ti.rescale_index(block2, block1, [3, 1]))       # output: [1, 0]
...
+        self.snode.place(self.pos)
+    @ti.kernel
+    def usage(self):
+        cnt = 0
+        for I in ti.grouped(self.snode):
+            if ti.is_active(self.snode, I):
+                cnt+=1
+        usage =  cnt/(grid_size[0]*grid_size[1])
+        print("Grid usage: ", usage)
+sp = SparseGrid(grid_size=grid_size)
```
Comment (ailzhang) [inline docs/lang/articles/basic/sparse.md:390]: @chunleili is it easier to just use ti.sparse.grid that you just added?

Comment (chunleili) [reply docs/lang/articles/basic/sparse.md:390]: Yes, but it is not released yet. So add the doc of this new API now may be confusing

Comment (ailzhang) [reply docs/lang/articles/basic/sparse.md:390]: @chunleili the doc you're modifying will be shown on a `develop` branch on the doc site (which tracks master branch so it's totally fine to use it here ;)

**Review by ailzhang — APPROVED** [review]
Thanks!

**Pull request merged**